relayer/relays/beefy/beefy-ethereum-listener.go

```diff
@@ -203,63 +272,79 @@ func (li *BeefyEthereumListener) processHistoricalInitialVerificationSuccessfulE
 					"initial_verification_tx": event.Raw.TxHash.Hex(),
 					"complete_on_block":       event.Raw.BlockNumber + li.blockWaitPeriod,
 				}
-				updateCmd := store.NewDatabaseCmd(item, store.Update, instructions)
-				li.dbMessages <- updateCmd
 
-				itemFoundInDatabase = true
+				select {
+				case <-ctx.Done():
+					return ctx.Err()
+				case li.dbMessages <- store.NewDatabaseCmd(item, store.Update, instructions):
+				}
+
+				found = true
 				break
 			}
 		}
-		if !itemFoundInDatabase {
+		if !found {
 			// Don't have an existing item to update, therefore we won't be able to build the completion tx
-			log.Error("BEEFY justification data not found in database for InitialVerificationSuccessful event. Ignoring event.")
+			return fmt.Errorf("item not found in database for InitialVerificationSuccessful event")
 		}
 	}
+
+	return nil
 }
 
 // processInitialVerificationSuccessfulEvents transitions matched database items from status
 // InitialVerificationTxSent to InitialVerificationTxConfirmed
-func (li *BeefyEthereumListener) processInitialVerificationSuccessfulEvents(ctx context.Context,
-	blockNumber uint64) {
-
+func (li *BeefyEthereumListener) processInitialVerificationSuccessfulEvents(
+	ctx context.Context,
+	blockNumber uint64,
+) error {
 	events, err := li.queryInitialVerificationSuccessfulEvents(ctx, blockNumber, &blockNumber)
 	if err != nil {
-		log.WithError(err).Error("Failure fetching event logs")
+		log.WithError(err).Error("Failure querying InitialVerificationSuccessful events")
+		return err
 	}
 
-	if len(events) > 0 {
-		log.Info(fmt.Sprintf("Found %d InitialVerificationSuccessful events on block %d", len(events), blockNumber))
-	}
+	log.WithFields(log.Fields{
+		"block": blockNumber,
+		"count": len(events),
+	}).Debug("Queried for InitialVerificationSuccessful events")
 
 	for _, event := range events {
 		log.WithFields(logrus.Fields{
 			"blockHash":   event.Raw.BlockHash.Hex(),
 			"blockNumber": event.Raw.BlockNumber,
 			"txHash":      event.Raw.TxHash.Hex(),
-		}).Info("event information")
+			"Prover":      event.Prover.Hex(),
+		}).Info("Processing InitialVerificationSuccessful event")
 
 		// Only process events emitted by transactions sent from our node
 		if event.Prover != li.ethereumConn.GetKP().CommonAddress() {
+			log.WithFields(logrus.Fields{
+				"Prover": event.Prover.Hex(),
+			}).Info("Skipping InitialVerificationSuccessful event as it has an unknown prover address")
 			continue
 		}
 
-		item := li.beefyDB.GetItemByInitialVerificationTxHash(event.Raw.TxHash)
-		if item.Status != store.InitialVerificationTxSent {
-			continue
+		item, err := li.beefyDB.GetItemByInitialVerificationTxHash(event.Raw.TxHash)
+		if err != nil {
+			log.Error("Failed to retrieve item from Beefy DB")
+			return err
```

Comment:
again, not sure if we want to actually have an error here, or just ignore and continue?

Reply:
fixed